modules/seqexec/engine/src/main/scala/seqexec/engine/Engine.scala

```diff
@@ -83,7 +83,7 @@ class Engine[F[_]: MonadError[?[_], Throwable]: Logger, S, U](stateL: Engine.Sta
     val x = for {
       seq <- stateL.sequenceStateIndex(c.sid).getOption(st)
       if (seq.status.isIdle || seq.status.isError) && !seq.getSingleState(c.actCoords).active
-      act <- seq.getSingleAction(c.actCoords)
+      act <- seq.rollback.getSingleAction(c.actCoords)
```

Comment:
This was a bit of a bug. Before, it could have happened that `startSingle` was called on a partially run step. It happened to worked OK. But it would not work properly with the implementation of the system overrides. That is because the update method in Step only updates the pending `Execution`s, not the run ones or the `focus`. But it properly updates the backup copy of the actions, so calling `rollback` gets the updated actions.